include/swift/Driver/Util.h

```diff
@@ -46,7 +46,11 @@ namespace driver {
     enum class WhichFiles : unsigned {
       Input,
       PrimaryInputs,
-      Output
+      Output,
+      /// Used in batch mode to provide supplementary output file names.
+      /// An oddball because there are lots of paths and
+      /// types, uses the whole DerivedOutputFileMap.
+      SupplementaryOutput,
```

Comment:
David and I have gone back and forth about this. It would mean switching WMO over to using the output file map as well, which is a little weird because WMO only has one of each supplementary output.

Reply:
I would like to subsume *all* filename arguments to the frontend in one file map, and use helpers for testing. I think the frontend code might be more maintainable with only one way to do things. Jordan & I have talked about this, but not very much. Not the fish to fry right now.